planner/core/exhaust_physical_plans.go

```diff
@@ -387,6 +387,93 @@ func (p *LogicalJoin) constructIndexJoin(
 	return []PhysicalPlan{join}
 }
 
+func findIndexScanAndTableScan(p PhysicalPlan) (is *PhysicalIndexScan, ts *PhysicalTableScan) {
+	var children []PhysicalPlan
+	switch x := p.(type) {
+	case *PhysicalTableReader:
+		children = x.TablePlans
+	case *PhysicalIndexReader:
+		children = x.IndexPlans
+	case *PhysicalIndexLookUpReader:
+		children = append(x.IndexPlans, x.TablePlans...)
+	default:
+		children = x.Children()
+	}
+	for _, child := range children {
+		tmpIS, tmpTS := findIndexScanAndTableScan(child)
+		if tmpIS != nil {
+			is = tmpIS
+		}
+		if tmpTS != nil {
+			ts = tmpTS
+		}
+	}
+	if tmpIS, ok := p.(*PhysicalIndexScan); ok {
+		return tmpIS, ts
+	}
+	if tmpTS, ok := p.(*PhysicalTableScan); ok {
+		return is, tmpTS
+	}
+	return is, ts
+}
+
+func (p *LogicalJoin) constructIndexMergeJoin(
+	prop *property.PhysicalProperty,
+	outerIdx int,
+	innerPlan PhysicalPlan,
+	ranges []*ranger.Range,
+	keyOff2IdxOff []int,
+	lens []int,
+	compareFilters *ColWithCmpFuncManager,
+) []PhysicalPlan {
+	indexJoins := p.constructIndexJoin(prop, outerIdx, innerPlan, ranges, keyOff2IdxOff, lens, compareFilters)
+	indexMergeJoins := make([]PhysicalPlan, 0, len(indexJoins))
+	for _, plan := range indexJoins {
+		is, ts := findIndexScanAndTableScan(innerPlan)
+		join := plan.(*PhysicalIndexJoin)
+		// needOuterSort means whether the outer join keys are the prefix of the prop items.
+		needOuterSort := len(prop.Items) > 0 && len(join.OuterJoinKeys) <= len(prop.Items)
```

Comment:
`len(join.OuterJoinKeys) <= len(prop.Items)` is sufficient?

Reply:
Yes, `prop.Items` are always satisfied by `outerChild`. The `needOuterSort` means whether outer rows should be sorted for building range.